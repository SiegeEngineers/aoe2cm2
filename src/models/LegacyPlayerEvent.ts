import LegacyCivilisation from "./LegacyCivilisation";
import ActionType from "../constants/ActionType";
import Player from "../constants/Player";

class LegacyPlayerEvent {
    public readonly player: Player;
    public readonly executingPlayer: Player;
    public readonly actionType: ActionType;
    public civilisation: LegacyCivilisation;
    public offset: number;

    constructor(player: Player, actionType: ActionType, civilisation: LegacyCivilisation, executingPlayer: Player = player) {
        this.player = player;
        this.executingPlayer = executingPlayer;
        this.actionType = actionType;
        this.civilisation = civilisation;
        this.offset = 0;
    }

    public static from(source: LegacyPlayerEvent): LegacyPlayerEvent {
        const playerEvent = new LegacyPlayerEvent(source.player, source.actionType, source.civilisation, source.executingPlayer);
        playerEvent.offset = source.offset;
        return playerEvent;
    }
}

export default LegacyPlayerEvent;
