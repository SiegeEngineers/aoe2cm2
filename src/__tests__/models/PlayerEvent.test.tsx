import LegacyPlayerEvent from "../../models/LegacyPlayerEvent";

it('LegacyPlayerEvent can deserialize old state', () => {
    const source = JSON.parse('{"player": "HOST", "actionType": "PICK", "civilisation": {}, "offset": 123}') as LegacyPlayerEvent;
    expect(LegacyPlayerEvent.from(source)).toMatchSnapshot();
});

it('LegacyPlayerEvent can deserialize new state', () => {
    const source = JSON.parse('{"player": "HOST", "actionType": "PICK", "civilisation": {}, "offset": 123, "executingPlayer": "GUEST"}') as LegacyPlayerEvent;
    expect(LegacyPlayerEvent.from(source)).toMatchSnapshot();
});
