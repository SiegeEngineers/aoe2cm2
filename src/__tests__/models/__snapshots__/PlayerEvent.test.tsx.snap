// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegacyPlayerEvent can deserialize new state 1`] = `
LegacyPlayerEvent {
  "actionType": "PICK",
  "civilisation": Object {},
  "executingPlayer": "GUEST",
  "offset": 123,
  "player": "HOST",
}
`;

exports[`LegacyPlayerEvent can deserialize old state 1`] = `
LegacyPlayerEvent {
  "actionType": "PICK",
  "civilisation": Object {},
  "executingPlayer": "HOST",
  "offset": 123,
  "player": "HOST",
}
`;
